pkgs/by-name/mf/mfem/package.nix

```diff
@@ -0,0 +1,74 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  gfortran,
+  mpi,
+  mpiSupport ? true,
+  mpiCheckPhaseHook,
+  blas,
+  hypre,
+  metis,
+  suitesparse,
+  withSuiteSparse ? mpiSupport,
+  mumps_par,
+  withMumps ? mpiSupport,
+}:
+assert withSuiteSparse -> mpiSupport;
+assert withMumps -> mpiSupport;
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mfem";
+  version = "4.7";
+
+  src = fetchFromGitHub {
+    owner = "mfem";
+    repo = "mfem";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-NJI8ABsfnnzMETfyscpNX8qK72BFF9FfxsY0J48aRtg=";
+  };
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [
+    cmake
+    gfortran
+  ] ++ lib.optional mpiSupport mpi;
+
+  buildInputs =
+    lib.optionals mpiSupport [
+      blas
+      hypre # mpi->hypre
+      metis # hypre->metis
+    ]
+    ++ lib.optional withSuiteSparse suitesparse
+    ++ lib.optionals withMumps ([ mumps_par ] ++ mumps_par.buildInputs);
```

Comment:
Maybe we need to modify `mumps_par` and make its `buildInputs` `propagatedBuildInputs`

Reply:
when built withMumps, cmake rule force to require (blas lapack scalapack metis and parmetis), it just treat mumps_par as a collection of packages, since mumps_par lately merged into nixpkgs and for now only mfem requires on it, the way how mumps serve as a dependency library may differ.